### ISSUE TITLE
No warning about using * operator on sequence types containing objects

### ISSUE THREAD

**Issue by carlio** [pylint-dev/pylint#8779]
### Bug description

This is not so much a bug but a missing check, perhaps.

Given the following file called 'test.py'

```
a = [[]]*2
print(a[0] is a[1])
b = [[],[]]
print(b[0] is b[1])
```

The output is:

```
a = [[]]*2
print(a[0] is a[1])
b = [[],[]]
print(b[0] is b[1])
```

I recently had a bug which because I did not realise that using `*2` would create a copy of the pointer to the initial sub-list, instead of creating a new list. Pylint does not detect this.

I'm not sure what this error message would be called or the exact phrasing.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:5:0: C0305: Trailing newlines (trailing-newlines)
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 0.00/10, +5.00)
```

### Expected behavior

Ideally, pylint would warn that using `*` on a sequence type containing objects does not create new objects but instead refers to the same object. In [the documentation](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range):

`s * n or n * s` is equivalent to "equivalent to adding s to itself n times" - which significantly means the list has the same element added to itself multiple times, This does not matter for primitives but for objects it does.

### Pylint version

```shell
$ pylint --version
pylint 2.17.4
astroid 2.15.5
Python 3.11.0 (main, Oct 25 2022, 13:54:35) [GCC 9.4.0]
```

**Comment by carlio** [comment]
I would consider this to be a false-negative, perhaps. I'm also happy to be assigned this issue, but I would require some help knowing exactly what kind of phrasing to put into the error message.

Just to further show how a bug can manifest itself when using this:

```a = [[]]*2
print(a)
a[0].append(1)
print(a)
```

when run:

```$ python test.py 
[[], []]
[[1], [1]]
```

**Comment by Pierre-Sassoulas** [comment]
Sounds like a nice check !

If I understood correctly it would be raised only for non primitive type (not for ``"n" * 50``) and only if the result can be reused later (not for ``", ".join(["n"] * 50)``). Regarding the symbolic name, what about ``shared-reference-via-multiplication``, ``multiplication-shallow-copying`` or ``duplicate-reference-in-collection`` ? And for the message itself maybe: "Consider using explicit comprehension or other methods for creating independent copies" ?
